### DISCUSSION TITLE
Is there a way to "move" some libs to different path using cx_freeze?

### DISCUSSION THREAD

**Discussion by itsMGA** [marcelotduarte/cx_Freeze#2401] [Q&A]
So I m using cx_freeze to generate .exe
Main problem occurses with some of the lib/pyc files, they conflic with my other app pyc's I saw there was an zip_include_packages option, that succesfuly moved/zipped encodings package

Problem remains with some __init__.py's and pyexpat.pyd these I dont know how to attempt to zip or move or whatever but damn its killing me at this point

Cx_freeze seems to be the last option to generate a py exe that is not a virus, and I do not indent to bundle everything into an exe at all

Do u guys know a way to deal with conflicting libs or rename that darn "lib" dir which is the most generic thing ever...? 

```
import sys, os
from cx_Freeze import setup, Executable

# Dependencies are automatically detected, but it might need fine tuning.
build_exe_options = {
    "packages": ["pyarmor", "os", "sys", "hashlib", "json", "time", "logging", "subprocess", "traceback", "paramiko", "PyQt5"],
    "include_files": [("dist/pyarmor_runtime_000000", "dist/pyarmor_runtime_000000")],
    # 'zip_includes': [("lib/pyexpat.pyd", "lib/pyexpat.pyd")]
}

base = None
if sys.platform == "win32":
    base = "Win32GUI"

# Define the executable
executables = [
    Executable("dist/patcher-source.py", base=base, target_name="Awakening2-Patcher.exe", icon='patcher-icon.ico')
]

setup(
    name="Awakening2-Patcher",
    version="1.0",
    description="Auto Patcher for Awakening2",
    options={"build_exe": build_exe_options},
    executables=executables
)
```

**Comment by itsMGA** [comment]
Help a brother out, been wasting so much time on this damn patcher thing, and just to not test it till now and found out in cooked and can't really use it :(

**Reply by marcelotduarte** [reply]
zip_include_packages only work for .py files of the packages (not modules), that are included as .pyc files. The .pyd files stay out of it. If a package 'package' has a module 'module.pyd', a file named 'package.module.pyd' was created in the lib directory and the others files go to library.zip.

I do not understand why you had conflicts with pyexpat. Do you a module of you application named pyexpat? If so, you can change the name of your module.
If you have problems updating the application and pyexpact is also being updated... well, who knows, pyexpact is not used by your program, it is just being included because it is part of the stdlib. So in this case, you can use `excludes = ["pyexpact"]`.

**Reply by itsMGA** [reply]
I understand, well pyexpact is not the only one, thing is there' s another executable, legacy one, p old, that is generated via cpp, so it also generates its libs and lib directory, both my exec and that one sit in the same parent dir, so the conflic happens on lib/*some.pyc files, ideally, just having the option to decide where to put lib and what not would solve all this really, or something that would help me work around it, there are lots of software that store libs within lib dir, which might be tricky to manage

**Reply by marcelotduarte** [reply]
How you execute this legacy executable? Do you use subprocess.run?

**Reply by itsMGA** [reply]
So the idea is:

There is a game launcher, so starter.exe 
And I made a patcher, that is executed occasionally to get updates, and trigger the starter.exe using os.startfile()

I could try to move the starter from main dir, but might cause problems with other file dependencies that are statically read within the legacy launcher, hmm

Still I'd prefer not to, it has dll's and reads lots of data from pack files, i essentially have to have something like from:
/game -> /game/patcher/ and game/source-game, which isn't ideal atm
